samples/boards/stm32/bluetooth/interactive_gui/src/main.c

```diff
@@ -0,0 +1,473 @@
+/*
+ * Copyright (c) 2024 STMicroelectronics
+ * Copyright (c) 2016 Nordic Semiconductor ASA
+ * Copyright (c) 2015-2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/drivers/bluetooth/hci_driver_bluenrg.h>
+#include <zephyr/bluetooth/hci_types.h>
+#include <zephyr/usb/usb_device.h>
+#include <zephyr/net/buf.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/l2cap.h>
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/bluetooth/buf.h>
+#include <zephyr/bluetooth/hci_raw.h>
+#include <version.h>
+
+#define LOG_MODULE_NAME gui_hci_uart
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+static const struct device *const hci_uart_dev =
+	DEVICE_DT_GET(DT_CHOSEN(zephyr_bt_c2h_uart));
+static K_THREAD_STACK_DEFINE(tx_thread_stack, CONFIG_BT_HCI_TX_STACK_SIZE);
+static struct k_thread tx_thread_data;
+static K_FIFO_DEFINE(tx_queue);
+
+/* RX in terms of bluetooth communication */
+static K_FIFO_DEFINE(uart_tx_queue);
+
+#define H4_CMD		0x01
+#define H4_ACL		0x02
+#define H4_SCO		0x03
+#define H4_EVT		0x04
+#define H4_ISO		0x05
```

Comment:
These should use the existing defines in `include/zephyr/bluetooth/hci_types.h` instead of redefining them here (IIRC they're called `BT_HCI_H4_*` there.

Reply:
...and you're already including `hci_types.h` so it should be a trivial change

Reply:
Done.